src/Router.php

```diff
@@ -39,4 +40,20 @@ public function routeMainController(MainController $mainController)
 
         return $this;
     }
+
+
+    /**
+     * @param ThreadController $threadController
+     * @return $this
+     */
+    public function routePostController(ThreadController $threadController){
+        $this->application->get('/threads/new', function(Request $request) use ($threadController) {
+            return $threadController->newThreadAction($request);
+        });
+
+        $this->application->post('/threads/new', function(Request $request) use ($threadController) {
+            return $threadController->createNewThreadAction($request);
+            //return $this -> application ->redirect('/', 302);
+        });
```

Comment:
docblock says that you are returning $this but you actually return void. It would be useful to return $this. See above comment.